project/default.nix

```diff
@@ -133,30 +140,78 @@ let
   overrides' = nixpkgs.lib.composeExtensions
     (self: super: mapAttrs (name: path: self.callCabal2nix name path {}) packages) overrides;
   mkPkgSet = name: _: this.${name}.override { overrides = overrides'; };
-in makeExtensible (prj: mapAttrs mkPkgSet shells // {
-  shells = mapAttrs (name: pnames:
-    this.workOnMulti' {
-      env = prj.${name}.override { overrides = self: super: nixpkgs.lib.optionalAttrs withHoogle {
-        ghcWithPackages = self.ghcWithHoogle;
-      }; };
-      packageNames = pnames;
-      inherit tools;
-    }
-  ) shells;
-
-  android = mapAttrs (name: config:
-    let
-      ghcAndroidArm64 = this.ghcAndroidArm64.override { overrides = overrides'; };
-      ghcAndroidArmv7a = this.ghcAndroidArmv7a.override { overrides = overrides'; };
-    in (this.androidWithHaskellPackages { inherit ghcAndroidArm64 ghcAndroidArmv7a; }).buildApp
-      ({ package = p: p.${name}; } // config)
-  ) android;
-
-  ios = mapAttrs (name: config:
-    let ghcIosArm64 = this.ghcIosArm64.override { overrides = overrides'; };
-    in (this.iosWithHaskellPackages ghcIosArm64).buildApp
-      ({ package = p: p.${name}; } // config)
-  ) ios;
-
-  reflex = this;
-})
+  prj = mapAttrs mkPkgSet shells // {
+    shells = mapAttrs (name: pnames:
+      this.workOnMulti' {
+        env = prj.${name}.override { overrides = self: super: nixpkgs.lib.optionalAttrs withHoogle {
+          ghcWithPackages = self.ghcWithHoogle;
+        }; };
+        packageNames = pnames;
+        inherit tools;
+      }
+    ) shells;
+
+    android =
+      mapAttrs (name: config:
+        let
+          ghcAndroidArm64 = this.ghcAndroidArm64.override { overrides = overrides'; };
+          ghcAndroidArmv7a = this.ghcAndroidArmv7a.override { overrides = overrides'; };
+        in (this.androidWithHaskellPackages { inherit ghcAndroidArm64 ghcAndroidArmv7a; }).buildApp
+          ({ package = p: p.${name}; } // config)
+      ) android;
+
+    ios =
+      mapAttrs (name: config:
+        let ghcIosArm64 = this.ghcIosArm64.override { overrides = overrides'; };
+        in (this.iosWithHaskellPackages ghcIosArm64).buildApp
+          ({ package = p: p.${name}; } // config)
+      ) ios;
+
+    reflex = this;
+
+    all = all true;
+  };
+
+  ghcLinks = mapAttrsToList (name: pnames: optionalString (pnames != []) ''
+    mkdir -p $out/${name}
+    ${concatMapStringsSep "\n" (n: ''
+      ln -s ${prj.${name}.${n}} $out/${name}/${n}
```

Comment:
I'd recommend putting single quotes around these arguments in case `name` has spaces. Other places as well.

Reply:
I think it would actually be better to use nixpkgs.lib.escapeShellArgs

Reply:
Oh! I wasn't aware of that function. Nice.

Reply:
Well `name` should be a GHC name from `this` and `n` should be a haskell package name, so neither of those should ever have spaces.

Reply:
Done anyway :)